develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown

```diff
@@ -0,0 +1,426 @@
+
+        public class UpgradePortletSettings
+                extends com.liferay.portal.upgrade.v7_0_0.UpgradePortletSettings {
+
+                public UpgradePortletSettings(SettingsFactory settingsFactory) {
+                        super(settingsFactory);
+                }
+
+                @Override
+                protected void doUpgrade() throws Exception {
+                        DLGroupServiceSettings.registerSettingsMetadata();
+                        DLPortletInstanceSettings.registerSettingsMetadata();
+
+                        upgradeMainPortlet(
+                                DLPortletKeys.DOCUMENT_LIBRARY, DLConstants.SERVICE_NAME,
+                                PortletKeys.PREFS_OWNER_TYPE_GROUP, true);
+
+                        upgradeDisplayPortlet(
+                                DLPortletKeys.DOCUMENT_LIBRARY, DLConstants.SERVICE_NAME,
+                                PortletKeys.PREFS_OWNER_TYPE_LAYOUT);
+                        upgradeDisplayPortlet(
+                                DLPortletKeys.MEDIA_GALLERY_DISPLAY, DLConstants.SERVICE_NAME,
+                                PortletKeys.PREFS_OWNER_TYPE_LAYOUT);
+                }
+
+        }
+
+    If you examine the `UpgradePortletSettings` class, you'll see that it
+    extends the
+    [com.liferay.portal.upgrade.v7_0_0.UpgradePortletSettings](https://github.com/liferay/liferay-portal/blob/692cff9635bd909ba13fab22574d4c02bae3ba91/portal-impl/src/com/liferay/portal/upgrade/v7_0_0/UpgradePortletSettings.java)
+    class, which extends the `UpgradeProcess` class. This follows the old
+    upgrade framework.
+
+    new framework [DLWebUpgrade.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/collaboration/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upgrade/DLWebUpgrade.java):
+
+        @Component(immediate = true, service = UpgradeStepRegistrator.class)
+        public class DLWebUpgrade implements UpgradeStepRegistrator {
+
+                @Override
+                public void register(Registry registry) {
+                        registry.register(
+                                "com.liferay.document.library.web", "0.0.0", "1.0.0",
+                                new DummyUpgradeStep());
+
+                        registry.register(
+                                "com.liferay.document.library.web", "0.0.1", "1.0.0",
+                                new UpgradeAdminPortlets(),
+                                new UpgradePortletSettings(_settingsFactory));
+        }
+
+    As you can see the new framework calls the old `UpgradePortletSettings`
+    framework class and uses it in the new registration method without any
+    problems.
+
+    The `DLWebUpgrade` new framework example covers two different use cases with 
+    two different registrations. If the first registration is not needed, the 
+    upgrade process skips it and uses the second upgrade registration instead.
```

Comment:
It's important to know that we are registering upgrade steps, not executing them. The framework will internally decide if it should be executed or not.